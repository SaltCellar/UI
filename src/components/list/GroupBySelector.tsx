import { Select, SelectOption, Title } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';
import React from 'react';
import { noop } from '../../utils/noop';
import { styles } from './header.styles';

interface GroupByOption {
  value: string;
  label: string;
}

interface GroupByProps {
  title: string;
  label: string;
  onChange: (value) => void;
  options?: GroupByOption[];
  initValue?: string;
}

export const GroupBySelector: React.SFC<GroupByProps> = ({
  title,
  label,
  onChange,
  options,
  initValue,
}) => (
  <React.Fragment>
    <Title size="2xl">{title}</Title>
    <div className={css(styles.groupBySelector)}>
      <label className={css(styles.groupBySelectorLabel)}>{label}</label>
    </div>
    <Select
      onBlur={noop}
      onFocus={noop}
      value={initValue}
      aria-label="group by selector"
      onChange={onChange}
    >
      {options.map(option => (
        <SelectOption
          key={`group-by-${option.value}`}
          value={option.value}
          label={option.label}
        />
      ))}
    </Select>
  </React.Fragment>
);

GroupBySelector.defaultProps = {
  options: [],
  initValue: '',
};
