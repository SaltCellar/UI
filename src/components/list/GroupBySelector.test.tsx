import { Title } from '@patternfly/react-core';
import * as emotion from 'emotion';
import { shallow } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import React from 'react';
import { GroupBySelector } from './GroupBySelector';

expect.addSnapshotSerializer(createSerializer(emotion));

test('render GroupBySelector without options', () => {
  const view = shallow(
    <GroupBySelector
      title="My Charges"
      label="GROUP CHRGS BY"
      onChange={jest.fn()}
    />
  );
  expect(view).toMatchSnapshot();
});

test('render GroupBySelector with options', () => {
  const view = shallow(
    <GroupBySelector
      title="My Charges"
      label="GROUP CHRGS BY"
      onChange={jest.fn()}
      initValue="initial value!"
      options={[{ value: '14', label: '14th' }, { value: '21', label: '21st' }]}
    />
  );
  expect(view).toMatchSnapshot();
});
