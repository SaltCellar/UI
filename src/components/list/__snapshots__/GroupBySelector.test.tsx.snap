// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GroupBySelector with options 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  margin-bottom: 0;
  margin-right: var(--pf-global--spacer--md);
}

<React.Fragment>
  <Title
    className=""
    size="2xl"
  >
    My Charges
  </Title>
  <div
    className="emotion-1"
  >
    <label
      className="emotion-0"
    >
      GROUP CHRGS BY
    </label>
  </div>
  <Select
    aria-label="group by selector"
    className=""
    isDisabled={false}
    isValid={true}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    value="initial value!"
  >
    <SelectOption
      className=""
      isDisabled={false}
      key="group-by-14"
      label="14th"
      value="14"
    />
    <SelectOption
      className=""
      isDisabled={false}
      key="group-by-21"
      label="21st"
      value="21"
    />
  </Select>
</React.Fragment>
`;

exports[`render GroupBySelector without options 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  margin-bottom: 0;
  margin-right: var(--pf-global--spacer--md);
}

<React.Fragment>
  <Title
    className=""
    size="2xl"
  >
    My Charges
  </Title>
  <div
    className="emotion-1"
  >
    <label
      className="emotion-0"
    >
      GROUP CHRGS BY
    </label>
  </div>
  <Select
    aria-label="group by selector"
    className=""
    isDisabled={false}
    isValid={true}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onFocus={[Function]}
    value=""
  />
</React.Fragment>
`;
